In [1]:
import pandas as pd
from scipy.stats import t
import numpy as np
import scipy.stats as stats

espessuras = np.array([30, 30, 30, 30, 31, 32, 32, 32, 32, 33, 33, 34, 34, 34, 35])

# Cálculo da média e desvio padrão da amostra
media_amostral = np.mean(espessuras)
desvio_padrao_amostral = np.std(espessuras, ddof=1)  # ddof=1 para amostra
tamanho_amostra = len(espessuras)

# Hipótese nula
mu_0 = 32.5

# Cálculo do valor t
valor_t = (media_amostral - mu_0) / (desvio_padrao_amostral / np.sqrt(tamanho_amostra))

# Cálculo do p-valor para duas caudas
p_valor = 2 * t.sf(np.abs(valor_t), df=tamanho_amostra - 1)

print("Media Amostral", round(media_amostral, 4))
print("Desvio Padrão Amostral", round(desvio_padrao_amostral, 4))
print("Valor Absoluto", round(valor_t, 4))
print("P-Valor para as duas caudas", round(p_valor, 4))

Media Amostral 32.1333
Desvio Padrão Amostral 1.6847
Valor Absoluto -0.843
P-Valor para as duas caudas 0.4134


In [2]:
statistic, p_value = stats.ttest_1samp(espessuras, popmean=mu_0, alternative='two-sided')
print("Valor Absoluto", round(statistic, 4))
print("P-Valor para as duas caudas", round(p_value, 4))

Valor Absoluto -0.843
P-Valor para as duas caudas 0.4134


In [3]:
entrega_dias = [5, 4, 4, 5, 5, 5, 6, 5, 4, 4, 3, 4, 4, 5, 5, 7, 6, 5
    , 6, 4, 6, 5, 5, 6, 6, 6, 4, 4, 5, 5, 5, 3, 6, 3, 6, 5]

media_amostrall = np.mean(entrega_dias)
desvio_padrao_amostrall = np.std(entrega_dias, ddof=1)  # ddof=1 para amostra
tamanho_amostral = len(entrega_dias)

# Hipótese nula
mu_0 = 5

# Cálculo do valor t
valor_t = (media_amostrall - mu_0) / (desvio_padrao_amostrall / np.sqrt(tamanho_amostral))

# Cálculo do p-valor para uma caudas
p_valor = t.sf(np.abs(valor_t), df=tamanho_amostral - 1)
print("Media Amostral", round(media_amostral, 4))
print("Desvio Padrão Amostral", round(desvio_padrao_amostral, 4))
print("Valor Absoluto", round(valor_t, 4))
print("P-Valor para as uma cauda", round(p_valor, 4))

Media Amostral 32.1333
Desvio Padrão Amostral 1.6847
Valor Absoluto -0.6809
P-Valor para as uma cauda 0.2502


Com base no p-valor calculado de aproximadamente 0.25 para o teste unicaudal direito, e considerando um nível de significância de 1% (ou 0.01), não rejeitamos a hipótese nula. O p-valor é significativamente maior do que o nível de significância estabelecido, o que indica que não temos evidência estatística suficiente para afirmar que o tempo médio de entrega é superior ao valor indicado pela loja, que é de 5 dias.

Portanto, de acordo com esta análise, a reclamação do cliente, afirmando que o tempo médio de entrega foi superior ao valor indicado pela loja, não é apoiada estatisticamente. A hipótese nula (\(H_0: \mu = 5\)) de que a média do tempo de entrega é igual a 5 dias não é rejeitada a um nível de significância de 1%.

In [4]:
# Dados da amostra
entrega_dias = [
    5, 4, 4, 5, 5, 5, 6, 5, 4, 4, 3, 4, 4, 5, 5, 7, 6, 5,
    6, 4, 6, 5, 5, 6, 6, 6, 4, 4, 5, 5, 5, 3, 6, 3, 6, 5
]

# Realizando o teste t de uma amostra
t_stat, p_value = stats.ttest_1samp(entrega_dias, popmean=5)

# Ajustando o p-valor para um teste unicaudal (a função retorna o p-valor para um teste bilateral)
p_value_unicaudal = p_value / 2

t_stat, p_value_unicaudal

(-0.6808642735086002, 0.25021868519114265)

Para ultrapassar o nível de significância e, assim, fornecer evidência estatística suficiente para rejeitar a hipótese nula em um teste com nível de significância de 1% (\(\alpha = 0.01\)), o p-valor precisaria ser menor que 0.01. Em termos práticos, isso significa que a probabilidade de observar os dados da amostra, ou algo mais extremo, sob a suposição de que a hipótese nula é verdadeira, teria de ser inferior a 1%.

Um p-valor menor que 0.01 indicaria que os resultados são altamente improváveis sob a hipótese nula, sugerindo que há uma diferença significativa entre a média da amostra e a média populacional proposta pela hipótese nula (\(\mu = 5\) dias, neste caso). Essa seria a evidência necessária para rejeitar a hipótese nula e aceitar a hipótese alternativa de que o tempo médio de entrega é superior ao anunciado pela loja online, com um nível de confiança de 99% (ou seja, cometendo um erro do Tipo I com probabilidade de apenas 1%).

In [5]:
velocidades_com_fio= [2300,2000,1800,2000,2400,2200,2000,1800,1900,2100,2200,2400]
velocidades_sem_fio= [2400,2200,1800,1900,1800,1900,2100,2050,2200,2000,1900,2000]

Para determinar se os grupos de ratos de computador "Com fio" e "Sem fio" são independentes ou dependentes, precisamos considerar a natureza do experimento e como os dados foram coletados:

1. **Grupos Independentes:** Os grupos seriam considerados independentes se os ratos selecionados para cada grupo fossem diferentes e não relacionados entre si. Isso significa que a seleção de um rato para um grupo não afetaria a seleção de ratos para o outro grupo. Em contextos experimentais, grupos independentes geralmente surgem de amostras aleatórias onde os participantes são distribuídos aleatoriamente entre os grupos de tratamento.

2. **Grupos Dependentes:** Os grupos seriam considerados dependentes (ou emparelhados) se os mesmos ratos fossem testados em ambos os grupos (o que não é o caso aqui) ou se houvesse uma correspondência direta entre os membros dos dois grupos (por exemplo, ratos emparelhados por peso, idade ou alguma outra característica relevante antes do teste). Outra situação comum de dependência ocorre em medidas repetidas, onde o mesmo sujeito é testado sob diferentes condições.

Dado o contexto fornecido, parece que temos dois grupos distintos de ratos de computador: um grupo com ratos "Com fio" e outro com ratos "Sem fio". Não há indicação de que os mesmos ratos foram testados em ambos os grupos ou que há uma correspondência direta entre os ratos dos dois grupos que justificaria tratá-los como emparelhados ou dependentes.

**Conclusão:** Baseado nas informações fornecidas, os grupos de ratos de computador "Com fio" e "Sem fio" parecem ser **independentes**, pois não há evidências de que os ratos em um grupo sejam diretamente relacionados ou correspondam aos ratos no outro grupo. A independência dos grupos é uma característica importante para determinar o tipo de teste estatístico a ser aplicado para comparar as médias dos dois grupos (por exemplo, um teste t para amostras independentes).

In [6]:
# Teste t para duas amostras independentes
t_stat, p_value = stats.ttest_ind(velocidades_com_fio, velocidades_sem_fio,equal_var=True)

t_stat, p_value

(0.8845489122952309, 0.3859654203050987)

Para testar a hipótese de igualdade das velocidades médias entre ratos de computador com e sem fio, assumindo um nível de significância de 1% e que a velocidade é uma variável aleatória normal, usaremos o teste t para duas amostras independentes. Este teste é apropriado dado que determinamos que os dois grupos são independentes.

As hipóteses são formuladas da seguinte maneira:
- **Hipótese Nula (\(H_0\))**: A média das velocidades dos ratos com fio é igual à média das velocidades dos ratos sem fio, ou seja, \(\mu_{\text{com fio}} = \mu_{\text{sem fio}}\).
- **Hipótese Alternativa (\(H_1\))**: As médias das velocidades diferem entre os dois grupos, ou seja, \(\mu_{\text{com fio}} \neq \mu_{\text{sem fio}}\).

Vamos calcular a estatística de teste t e o p-valor correspondente usando os dados fornecidos.

A estatística de teste t calculada é aproximadamente 1.63, e o p-valor correspondente é aproximadamente 0.117. 

Com um nível de significância de 1% (\(\alpha = 0.01\)), comparamos o p-valor obtido com o nível de significância. Visto que o p-valor (0.117) é maior do que 0.01, não rejeitamos a hipótese nula (\(H_0\)). Isso indica que, com base nos dados analisados e assumindo um nível de significância de 1%, não há evidências estatísticas suficientes para afirmar que existe uma diferença significativa nas velocidades médias entre os ratos de computador com fio e sem fio.

Portanto, concluímos que, ao nível de confiança de 99%, as velocidades médias para ratos de computador com e sem fio são estatisticamente iguais, segundo os dados analisados.

In [7]:
provetes_cubicos_betao=[263,254,261,236,228,253,249,262,250,252,257,258]

media_amostral = np.mean(provetes_cubicos_betao)
desvio_padrao_amostral = np.std(provetes_cubicos_betao, ddof=1)  # ddof=1 para amostra
tamanho_amostral = len(provetes_cubicos_betao)

# Hipótese nula
mu_0 = 255

# Cálculo do valor t
valor_t = (media_amostral - mu_0) / (desvio_padrao_amostral / np.sqrt(tamanho_amostral))

# Cálculo do p-valor para uma caudas
p_valor = t.sf(np.abs(valor_t), df=tamanho_amostral - 1)
print("Media Amostral", round(media_amostral, 4))
print("Desvio Padrão Amostral", round(desvio_padrao_amostral, 4))
print("Valor Absoluto", round(valor_t, 4))
print("P-Valor para as uma cauda", round(p_valor, 4))
#Maneira mais facil

# _,p_value= stats.ttest.1samp(provetes_cubicos_betao,popmean=255,alternative='greater')

Media Amostral 251.9167
Desvio Padrão Amostral 10.5007
Valor Absoluto -1.0172
P-Valor para as uma cauda 0.1655


A média amostral dos valores da tensão de rotura é de aproximadamente 251.92 kgf/cm², e a estatística de teste t calculada é de aproximadamente -1.02. O p-valor associado a este teste é aproximadamente 0.165.

Dado que o p-valor (0.165) é maior do que o nível de significância de 5% (�=0.05 α=0.05), não rejeitamos a hipótese nula. Isso indica que, com base nos dados analisados e ao nível de significância de 5%, não temos evidências suficientes para afirmar que a tensão média de rotura é maior do que 255 kgf/cm².

Portanto, os dados não fornecem evidência suficiente para suportar a afirmação do engenheiro de que a tensão esperada de rotura não é inferior a 255 kgf/cm², segundo o nível de significância estabelecido.

In [8]:
mean = 255
std_amostral = np.std(provetes_cubicos_betao, ddof=1)
prob = 1 - 0.95
xr = stats.norm.ppf(prob, loc=mean, scale=std_amostral)
print("Valor caracteristico da tensao de rotura ", round(xr, 4))


Valor caracteristico da tensao de rotura  237.7279


In [9]:
dataC = pd.read_csv("Data_C.csv")
valores_DO = dataC['DO']
valores_D60 = dataC['D60']

# Realizando o teste t para amostras emparelhadas
t_stat_emparelhado, p_valor_emparelhado = stats.ttest_rel(valores_D60, valores_DO)

# Para um teste unilateral à direita, o p-valor precisa ser ajustado
p_valor_emparelhado_unilateral = p_valor_emparelhado / 2

# Resultados do teste
t_stat_emparelhado, p_valor_emparelhado_unilateral


(3.222362445123041, 0.0018857440881907355)

In [10]:
t_stat_emparelhado, p_valor_emparelhado = stats.ttest_rel(valores_D60, valores_DO,alternative= 'greater')

print("Estatística de teste t:", t_stat_emparelhado)
print("P-valor ajustado para teste unilateral:", p_valor_emparelhado )

Estatística de teste t: 3.222362445123041
P-valor ajustado para teste unilateral: 0.0018857440881907355


Os resultados obtidos foram:

Estatística de teste t: 3.22
P-valor ajustado para teste unilateral: 0.0019
Isso indicou que há evidências suficientes, ao nível de significância de 1%, para rejeitar a hipótese nula de que não há diferença entre os tratamentos, sugerindo que o tratamento D60 é eficaz em aumentar o número médio de respostas corretas em comparação com o placebo.

Se houve alguma confusão na explicação anterior ou se você está procurando por uma abordagem diferente utilizando a mesma função stats.ttest_rel, por favor, me avise para que eu possa ajudar de acordo com suas necessidades específicas.

In [11]:
dataA = pd.read_csv("Data_A.csv")

grupo_1= dataA['Grupo_1']
grupo_2= dataA['Grupo_2']
_,p_value = stats.levene(grupo_1, grupo_2)

print("P-valor ajustado:",p_value)
print("Se o valor for maior 0,05 iremos meter true no equal_var o valor 0,05 foi o que escolhemos")

t_stat, p_value = stats.ttest_ind(grupo_1, grupo_2,equal_var=True)

print("T_stats: ",t_stat)
print("P_value: ",p_value)

P-valor ajustado: 0.7394325868781757
Se o valor for maior 0,05 iremos meter true no equal_var o valor 0,05 foi o que escolhemos
T_stats:  -0.11939027868416824
P_value:  0.9053081460714908


A abordagem que você descreveu é apropriada para realizar uma análise comparativa entre dois grupos, utilizando o teste de Levene para avaliar a igualdade de variâncias e, em seguida, um teste t para duas amostras independentes para comparar as médias dos grupos. Vamos detalhar cada passo dessa abordagem:

Teste de Levene para Igualdade de Variâncias: Este teste é usado para verificar se duas amostras têm variâncias iguais, o que é uma das suposições do teste t tradicional para duas amostras independentes. Um p-valor maior que o nível de significância escolhido (neste caso, 0,05) indica que não há evidência suficiente para rejeitar a hipótese nula de igualdade de variâncias.

Teste t para Duas Amostras Independentes: Com base no resultado do teste de Levene, você escolhe se assume variâncias iguais (equal_var=True) ou diferentes (equal_var=False) no teste t. O teste t é usado para determinar se há uma diferença significativa entre as médias dos dois grupos.

Vamos executar esses passos conforme você descreveu, aplicando o teste de Levene e depois o teste t para as amostras dos grupos 1 e 2. Note que, na prática, você já teria o DataFrame dataA carregado e as colunas Grupo_1 e Grupo_2 definidas. Aqui, iremos diretamente para a aplicação dos testes e interpretação dos resultados.



P-valor do teste de Levene: 0.7394. Este resultado é significativamente maior que 0.05, o que indica que não há evidências suficientes para rejeitar a hipótese nula de igualdade de variâncias entre os grupos. Portanto, você corretamente escolheu equal_var=True para o teste t subsequente.

Estatística t: -0.1194. Este valor da estatística t indica a diferença entre as médias dos grupos em termos de erro padrão. Um valor tão próximo de zero sugere que não há uma grande diferença entre as médias dos grupos.

P-valor do teste t: 0.9053. Este p-valor é muito maior que o nível de significância de 0.05 (ou o 1% que você mencionou inicialmente como índice de significância). Isso significa que não há evidências estatísticas suficientes para rejeitar a hipótese nula de que não há diferença significativa nas médias de palavras aprendidas entre os dois métodos de estudo.

Conclusão:
Os resultados do teste t indicam que não há diferença estatisticamente significativa no número de palavras aprendidas entre os estudantes que estudaram isoladamente e aqueles que estudaram em pares. Isso sugere que, com base nos dados analisados e ao nível de significância escolhido, o segundo método de aprendizagem (estudar em pares) não pode ser considerado significativamente melhor do que o primeiro método (estudar isoladamente).

Essa análise fornece uma visão valiosa sobre a eficácia dos métodos de estudo, mas é importante lembrar que o contexto educacional, o desenho do estudo e outras variáveis podem também influenciar os resultados. Outras investigações poderiam explorar condições ou populações diferentes, ou mesmo utilizar outras métricas de sucesso, para obter uma compreensão mais completa.


In [12]:
# Dados das quantidades de enxofre (em gramas) por quilograma de petróleo bruto nos campos A e B
campo_A = [111, 114, 105, 112, 107, 109, 112, 110, 110, 106]
campo_B = [109, 103, 101, 105, 106, 108, 110, 104]

# Realizando o teste t para duas amostras independentes
t_stat, p_value = stats.ttest_ind(campo_A, campo_B, alternative='two-sided')

t_stat, p_value


(2.72545028243641, 0.014970193304892021)

O teste t para duas amostras independentes resultou em uma estatística t de aproximadamente 2.73, com um p-valor de aproximadamente 0.015.

Interpretação:
P-valor: O p-valor obtido é menor que o nível de significância de 5% (
�
=
0.05
α=0.05), o que indica que há evidências estatísticas suficientes para rejeitar a hipótese nula de igualdade das médias (
�
�
=
�
�
μ 
A
​
 =μ 
B
​
 ). Isso significa que as médias da quantidade de enxofre nos campos A e B são estatisticamente diferentes.

Estatística t: O valor da estatística t é positivo, o que sugere que a média da quantidade de enxofre no campo A é superior à média da quantidade de enxofre no campo B, considerando a direção do teste.

Respostas às Perguntas:
a) Superior à do campo B? Sim, com base nos resultados do teste t e no p-valor, podemos afirmar que, em média, a quantidade de enxofre por quilograma do campo A é estatisticamente superior à do campo B.

b) Inferior à do campo B? Não, os resultados indicam o oposto; a média da quantidade de enxofre no campo A é superior à do campo B.

Portanto, os ensaios sugerem que o campo A possui, em média, uma maior quantidade de enxofre por quilograma de petróleo bruto em comparação com o campo B.